bentoml/artifact/tf_savedmodel_artifact.py

```diff
@@ -31,6 +31,79 @@ def _is_path_like(p):
     return isinstance(p, (str, bytes, pathlib.PurePath, os.PathLike))
 
 
+class _TensorflowFunctionWrapper:
+    '''
+    TensorflowFunctionWrapper
+    transform input tensor following function input signature
+    '''
+
+    def __init__(self, origin_func, fullargspec):
+        self.origin_func = origin_func
+        self.fullargspec = fullargspec
+        self._args_to_indices = {arg: i for i, arg in enumerate(fullargspec.args)}
+
+    def __call__(self, *args, **kwargs):
+        signatures = self.origin_func.input_signature
+        for k in kwargs:
+            if k not in self._args_to_indices:
+                raise TypeError(f"Function got an unexpected keyword argument {k}")
+        signatures_by_kw = {
+            k: signatures[self._args_to_indices[k]]
+            for k in kwargs
+        }
+        # INFO:
+        # how signature with kwargs works?
+        # https://github.com/tensorflow/tensorflow/blob/v2.0.0/tensorflow/python/eager/function.py#L1519
+
+        transformed_args = tuple(
+            self._transform_input_by_tensorspec(arg, signatures[i])
+            for i, arg in enumerate(args))
+        transformed_kwargs = {
+            k: self._transform_input_by_tensorspec(arg, signatures_by_kw[k])
+            for k, arg in kwargs.items()
+        }
+        return self.origin_func(*transformed_args, **transformed_kwargs)
+
+    def __getattr__(self, k):
+        return getattr(self.origin_func, k)
+
+    @staticmethod
+    def _transform_input_by_tensorspec(_input, tensorspec):
+        '''
+        transform dtype & shape following tensorspec
+        '''
+        try:
+            import tensorflow as tf
+        except ImportError:
+            raise ImportError(
```

Comment:
use `raise MissingDependencyException` instead? you may need to rebase on master

Reply:
ok